crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -114,15 +113,16 @@ impl UiSurface {
     }
 
     /// Update the children of the taffy node corresponding to the given [`Entity`].
-    pub fn update_children(&mut self, entity: Entity, children: &Children) {
-        let mut taffy_children = Vec::with_capacity(children.len());
+    pub fn update_children(&mut self, entity: Entity, children: impl Iterator<Item = Entity>) {
+        let mut taffy_children = Vec::new();
```

Comment:
```suggestion
        let mut taffy_children = std::mem::take(&mut self.taffy_children_scratch);
        taffy_children.clear();
```
Hmm can we move this vec into the `UiSurface` as `taffy_children_scratch`? Not a blocker for this PR but nice to have.

Reply:
Ahh right, now I see how it could be useful. In this case there was no need though.
I moved the vec and updated the code to use a borrow.

Thank you for the suggestion!